hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -221,9 +230,20 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
     TableOperations ops = newTableOps(identifier);
     TableMetadata lastMetadata = null;
-    if (purge) {
+
+    LOG.info("deleteTableRootDir: {}", this.deleteTableRootDir);
+    boolean externalTablePurge = false;
+    boolean gcEnabled = false;
+
+    if (purge || this.deleteTableRootDir) {
```

Comment:
Why do we delete the table root directory if the `purge` is `false`?
Is this the expected behavior?

Reply:
Maybe we should calculate the deleteData here, and then use it down the line.

Reply:
Consider it from the other perspective.
A Spark user is used to keep the table data when not using `purge` explicitly. The proposed behavior would be strange to them.

Would an auto-purge flag in Spark be better?